controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -577,30 +568,111 @@ func setAvailableCondition(_ context.Context, kcp *controlplanev1.KubeadmControl
 			controlplanev1.KubeadmControlPlaneMachineEtcdMemberHealthyV1Beta2Condition,
 			controlplanev1.KubeadmControlPlaneMachineEtcdPodHealthyV1Beta2Condition) {
 			k8sControlPlaneNotHealthy++
+		} else {
+			k8sControlPlaneNotHealthyButNotReportedYet++
 		}
 	}
 
+	// Maps machines to members
+	memberToMachineMap := map[string]*clusterv1.Machine{}
+	provisioningMachines := []*clusterv1.Machine{}
+	for _, machine := range machines {
+		if machine.Status.NodeRef == nil {
+			provisioningMachines = append(provisioningMachines, machine)
+			continue
+		}
+		for _, member := range etcdMembers {
+			if machine.Status.NodeRef.Name == member.Name {
+				memberToMachineMap[member.Name] = machine
+				break
+			}
+		}
+	}
+
+	// Determine etcd members availability by using etcd members as a source of truth because
+	// etcd members might not match with machines, e.g. while provisioning a new machine.
+	// Also in this case, we leverage info on machines to determine member health.
+	votingEtcdMembers := 0
+	learnerEtcdMembers := 0
+	etcdMembersHealthy := 0
+	etcdMembersNotHealthy := 0
+	etcdMembersNotHealthyButNotReportedYet := 0
+
+	for _, etcdMember := range etcdMembers {
+		if etcdMember.IsLearner || etcdMember.Name == "" {
```

Comment:
Are etcd members without name learner?

Reply:
Added a comment to explain why we are making this assumption